### ISSUE TITLE
When using a Lambda ARN I got "One or more Fn::Sub intrinsic functions don't specify expecte"

### ISSUE THREAD

**Issue by j0k3r** [horike37/serverless-step-functions#252]
# This is a Bug Report

## Description

Jumping from the version 2.1.0 to the 2.6.0, I converted all `Resource` from the full arn to the `Fn::getAtt` way:

```diff
-  Resource: arn:aws:lambda:#{AWS::Region}:#{AWS::AccountId}:function:${self:service}-${opt:stage}-scheduleContent
+  Resource:
+      Fn::GetAtt: [scheduleContent, Arn]
```

BUT I also have one Lambda ARN which isn't part of the serverless.yml but from an other serverless.yml. So I kept the old way to define the Resource: using the ARN:

```yaml
Type: Task
Resource: arn:aws:lambda:#{AWS::Region}:#{AWS::AccountId}:function:lyra-${opt:stage}-stats
```

(note: the _service_ isn't `self::service` but hardcoded to `lyra`, our other serverless project)

When deploying using serverless, I got an error when validating the template about the `Fn::Sub` intrinsic function:

>   Error: The CloudFormation template is invalid: Template error: One or more Fn::Sub intrinsic functions don't specify expected arguments. Specify a string as first argument, and an optional second argument to specify a mapping of values to replace in the string

Removing the ARN fixed the issue which confirm it's related to the full ARN in the `Resource`:

```diff
-Type: Task
+Type: Pass
-Resource: arn:aws:lambda:#{AWS::Region}:#{AWS::AccountId}:function:lyra-${opt:stage}-stats
```

Reverting back to the 2.1.0 with ARN fix the issue too.

## Additional Data
* ***Serverless Framework Core Version you're using***: 1.51.1 (confirmed to fail on 1.46.1 too)
* ***The Plugin Version you're using***: 2.6.0 (affected version starts at 2.2.0)
* ***Operating System***: MacOS
* ***Stack Trace***:
```
  Error: The CloudFormation template is invalid: Template error: One or more Fn::Sub intrinsic functions don't specify expected arguments. Specify a string as first argument, and an optional second argument to specify a mapping of values to replace in the string
      at provider.request.catch.error (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/lib/plugins/aws/deploy/lib/validateTemplate.js:19:13)
      at tryCatcher (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:517:31)
      at Promise._settlePromise (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:574:18)
      at Promise._settlePromise0 (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:619:10)
      at Promise._settlePromises (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:695:18)
      at _drainQueueStep (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:138:12)
      at _drainQueue (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:131:9)
      at Async._drainQueues (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:147:5)
      at Immediate.Async.drainQueues (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:17:14)
      at runCallback (timers.js:794:20)
      at tryOnImmediate (timers.js:752:5)
      at processImmediate [as _immediateCallback] (timers.js:729:5)

     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.
```
* ***Provider Error messages***:

**Comment by theburningmonk** [comment]
@jflewis can you paste your state machine definition in here?

**Comment by theburningmonk** [comment]
@j0k3r in this diff you shared
```
-Type: Task
+Type: Pass
-Resource: arn:aws:lambda:#{AWS::Region}:#{AWS::AccountId}:function:lyra-${opt:stage}-stats
```
The `Type` changed as well, could that have been the problem?

**Comment by j0k3r** [comment]
Hum nope, I put the type `Pass` just to keep the state (it already has the parameter `End: true`)

**Comment by theburningmonk** [comment]
Ok, in that case, can you paste the version that didn't work on here so I can debug it?

**Comment by j0k3r** [comment]
It only fails when I combine both solution:
- using shortcut `Fn::GetAtt: [lambdaName, Arn]` to retrieve the lambda ARN
- using the old way `arn:aws:lambda:#{AWS::Region}:#{AWS::AccountId}:function:${self:service}-${opt:stage}-lambdaName`

If I keep all `Resource` using the old way, it works. But as I need to use an Arn for a lambda defined in an other serverless project, I must combine both solution.

So here is a file that fail on Cloudformation validate:

```yml
stepFunctions:
    stateMachines:
        Push:
            name: Push
            definition:
                Comment: "A step function to push stuff"
                StartAt: PushFirstUsers
                States:
                    PushFirstUsers:
                        Type: Task
                        Resource:
                            Fn::GetAtt: [push, Arn]
                        Next: HasNextPage
                    HasNextPage:
                        Type: Choice
                        Choices:
                            -
                                Not:
                                    Variable: "$.cursor"
                                    BooleanEquals: false
                                Next: PushNextUsers
                        Default: PublishStats
                    PushNextUsers:
                        Type: Task
                        Resource:
                            Fn::GetAtt: [push, Arn]
                        Next: HasNextPage
                    PublishStats:
                        Type: Task
                        Resource: arn:aws:lambda:#{AWS::Region}:#{AWS::AccountId}:function:lyra-${opt:stage}-stats
                        End: true
```

And it fails at template validation:

```
Serverless: Uploading CloudFormation file to S3...
Serverless: Uploading artifacts...
Serverless: Uploading service project.zip file to S3 (8.89 MB)...
Serverless: Validating template...

  Error --------------------------------------------------

  Error: The CloudFormation template is invalid: Template error: One or more Fn::Sub intrinsic functions don't specify expected arguments. Specify a string as first argument, and an optional second argument to specify a mapping of values to replace in the string
      at provider.request.catch.error (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/lib/plugins/aws/deploy/lib/validateTemplate.js:19:13)
      at tryCatcher (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:517:31)
      at Promise._settlePromise (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:574:18)
      at Promise._settlePromise0 (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:619:10)
      at Promise._settlePromises (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:695:18)
      at _drainQueueStep (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:138:12)
      at _drainQueue (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:131:9)
      at Async._drainQueues (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:147:5)
      at Immediate.Async.drainQueues (/Users/jeremy/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:17:14)
      at runCallback (timers.js:794:20)
      at tryOnImmediate (timers.js:752:5)
      at processImmediate [as _immediateCallback] (timers.js:729:5)
```

**Comment by theburningmonk** [comment]
Thanks, I was able to reproduce the issue with your template. It looks like an incompatibility between `serverless-step-functions` and `serverless-pseudo-parameters` as both try to apply `Fn::Sub` to the state machine definition string.

The only workaround that I can think of is to apply the pseudo parameters in the step functions plugin as well. Not ideal, as we'd duplicate the functionality of another plugin.

@horike37 any thoughts on this?

**Comment by FilipPyrek** [comment]
@theburningmonk Hi, I have stumbled upon a same issue and removing  `serverless-pseudo-parameter` did the job. 👍

**Comment by j0k3r** [comment]
Thanks @theburningmonk, issue solved on our side with 2.10.0 👍
